components/provisioner/e2e_test/main_test.go

```diff
@@ -0,0 +1,165 @@
+			Message   string `json:"message"`
+			LastError struct {
+				ErrMessage string `json:"errMessage"`
+			} `json:"lastError"`
+		} `json:"lastOperationStatus"`
+	} `json:"runtimeStatus"`
+}
+
+type DeprovisionResp struct {
+	DeprovisionRuntime string `json:"deprovisionRuntime"`
+}
+
+type GQLClient struct {
+	reqsPath       string
+	providerSecret string
+	provider       string
+	client         *graphql.Client
+}
+
+func (gql GQLClient) gqlRequest(ctx context.Context, reqName string, vars map[string]string, resp interface{}) error {
+	payload, err := os.ReadFile(path.Join(gql.reqsPath, reqName))
+	if err != nil {
+		return err
+	}
+
+	//$name: String!, $provider: String!, $provider_secret: String
+	req := graphql.NewRequest(string(payload))
+
+	for key, val := range vars {
+		req.Var(key, val)
+	}
+
+	req.Header.Add("Content-Type", "application/json")
+	req.Header.Add("Tenant", tenant)
+
+	err = gql.client.Run(ctx, req, resp)
+
+	return err
+}
+
+func (gql GQLClient) provision(ctx context.Context, name, provider, provider_secret string) (resp ProvisionResp, err error) {
+	err = gql.gqlRequest(
+		ctx,
+		"provision.graphql",
+		map[string]string{"name": name, "provider": provider, "provider_secret": provider_secret},
+		&resp)
+	return
+}
+
+func (gql GQLClient) deprovision(ctx context.Context, runtimeID string) (resp DeprovisionResp, err error) {
+	err = gql.gqlRequest(
+		ctx,
+		"deprovision.graphql",
+		map[string]string{"runtimeID": runtimeID},
+		&resp)
+	return
+}
+
+func (gql GQLClient) status(ctx context.Context, runtimeID string) (resp StatusResp, err error) {
+	err = gql.gqlRequest(
+		ctx,
+		"status.graphql",
+		map[string]string{"runtimeID": runtimeID},
+		&resp)
+	return
+}
+
+func (gql GQLClient) waitForOp(ctx context.Context, runtimeID string) (resp StatusResp, err error) {
+	start := time.Now()
+	defer fmt.Println()
+	for {
+		resp, err = gql.status(ctx, runtimeID)
+		if err != nil {
+			return
+		}
+
+		msg := resp.RuntimeStatus.LastOperationStatus.Message
+
+		if strings.HasPrefix(msg, "Operation in progress.") ||
+			strings.HasSuffix(msg, "started") {
+			if time.Since(start) > waitTimeout {
+				return
+			}
+			fmt.Print(".")
+			time.Sleep(waitDelay)
+			continue
+		}
+
+		return
+	}
+}
+
+func TestName(t *testing.T) {
+	if os.Getenv("APP_GARDENER_KUBECONFIG_PATH") == "" {
+		t.SkipNow()
+	}
+
+	ctx := context.Background()
+
+	providerSecret := os.Getenv("PROVIDER_SECRET_NAME")
+	provider := os.Getenv("PROVIDER")
+	cli := GQLClient{
+		reqsPath:       "./requests/",
+		providerSecret: providerSecret,
+		provider:       provider,
+		client:         graphql.NewClient("http://localhost:3000/graphql"),
+	}
+
+	name := fmt.Sprintf("pts%d", time.Now().Unix())
+
+	t.Logf("Provisioning a %s cluster - %s", provider, name)
+	provisionResp, err := cli.provision(ctx, name, provider, providerSecret)
+	assert.NoError(t, err)
+	t.Log(provisionResp)
+
+	runtimeID := provisionResp.ProvisionRuntime.RuntimeID
```

Comment:
Writing myself a reminder to try to break something and see whether the test will be red.